### ISSUE TITLE
Feature request: Make `Detekt` and `DetektCreateBaselineTask` implement a common interface

### ISSUE THREAD

**Issue by ZacSweers** [detekt/detekt#5940]
Your feature request may already be reported!
Please search in the [issues](https://github.com/detekt/detekt/issues) before creating a new one.

## Expected Behavior

Currently, these two tasks are completely independent but have overlapping inputs. This means that if we want to configure Detekt (for example – `jvmTarget`), we must manually configure both types. It would be convenient if this shared configuration was extracted to a common interface (that extends `Task`) that could be used to configure both.

## Current Behavior

```kotlin
project.tasks.configureEach<Detekt> {
  this.jvmTarget = jvmTarget
  exclude("**/build/**")
}
project.tasks.configureEach<DetektCreateBaselineTask> {
  this.jvmTarget = jvmTarget
  exclude("**/build/**")
}
```

Desired, using a name like `DetektTask`

```kotlin
project.tasks.configureEach<DetektTask> {
  // Covers both Detekt and DetektCreateBaselineTask
  this.jvmTarget = jvmTarget
  exclude("**/build/**")
}
```

**Comment by atulgpt** [comment]
Hi @ZacSweers are you talking something similar to https://github.com/detekt/detekt/pull/5946 ?

**Comment by ZacSweers** [comment]
Yup! Note you can use the Task interface rather than DefaultTask and then you don't need to make the base task also a SourceTask (since the baseline task wasn't one before). Then the base task can also be an interface

**Comment by atulgpt** [comment]
`since the baseline task wasn't one before` it was `SourceTask` before as well. See below SS
<img width="1392" alt="Screenshot 2023-03-28 at 1 14 01 AM" src="https://user-images.githubusercontent.com/11026203/228049650-cc867eb8-f508-4e05-8616-cf628790a40a.png">
